### DISCUSSION TITLE
256 model produces poor results

### DISCUSSION THREAD

**Discussion by piddnad** [PixArt-alpha/PixArt-alpha#108] [Q&A] [answered]
Hello, thank you for doing such a great job! 

I tried PixArt-alpha-1024&512 and got really good results. 

However, when I go with PixArt-α-SAM-256 and PixArt-α-256, I find that the results tend to be fuzzy, like the following: 
![A small cactus with a happy face in the Sahara desert](https://github.com/PixArt-alpha/PixArt-alpha/assets/29009293/2f4f22de-6baa-410b-ab5c-cc4c33f875e3)

The way I load the 256px model is:
```
pipe = PixArtAlphaPipeline.from_pretrained("output/PixArt-XL-2-1024-MS", torch_dtype=torch.float16, use_safetensors=True)  # PixArt-α-1024
model = Transformer2DModel.from_pretrained("output/PixArt-XL-2-SAM-256x256", subfolder="transformer", torch_dtype=torch.float16, use_safetensors=True)
pipe.transformer = model
```

Is there something that I did wrong? Thanks in advance for your answers!

**Comment by lawrence-cj** [comment, marked as answer]
We have fixed the bug in the earlier diffusers version. Try to update the diffusers with `pip install -U diffusers`

**Comment by piddnad** [comment]
Thank you for your response – the issue is resolved! I think this indeed appears to be a bug with diffusers. 

I also saw a similar discussion here: https://github.com/huggingface/diffusers/issues/6783

In short, it seems that older versions of diffusers incorrectly applied the 512 size bucket (ASPECT_RATIO_512_BIN) for resolutions outside of 1024, thus leading to incorrect height and width values (see [here](https://github.com/huggingface/diffusers/blob/66f94eaa0c68a893b2aba1ec9f79ee7890786fba/src/diffusers/pipelines/pixart_alpha/pipeline_pixart_alpha.py#L776)). Upgrading diffusers resolves this issue.

Thanks again for your answer. :)

**Comment by lawrence-cj** [comment]
I have to mention that turning `use_resolution_binning = False` will cause unreasonable results when you specify a strange height/width for output.

**Reply by piddnad** [reply]
Got it! I revised my previous reply.
